EKF/mag_fusion.cpp

```diff
@@ -44,6 +44,18 @@
 #include <ecl.h>
 #include <mathlib/mathlib.h>
 
+Dual4f Ekf::yaw321FromQuaterion(matrix::Quaternion<Dual4f> q) const
+{
+	return atan2(Dual4f(2) * (q(0) * q(3) + q(1) * q(2)),
```

Comment:
```suggestion
	return atan2f(Dual4f(2) * (q(0) * q(3) + q(1) * q(2)),
```

On a board without double precision floating point (eg pixhawk 1, 2, 3, pixracer, etc) atan2 is an order of mangitude slower than atan2f.

``` Console
atan2(): 1000 events, 27911us elapsed, 27.91us avg, min 27us max 28us 0.284us rms
atan2f(): 1000 events, 2725us elapsed, 2.73us avg, min 2us max 3us 0.446us rms
```

http://ci.px4.io:8080/blue/organizations/jenkins/PX4_misc%2FFirmware-hardware/detail/master/1271/pipeline/788#step-800-log-361

Reply:
Good catch. @kamilritz this is probably the majority of the extra flash size you are seeing, too.

Reply:
Exactly